src/test/java/com/iexec/worker/replicate/ReplicateRecoveryServiceTests.java

```diff
@@ -54,11 +57,122 @@
     @Mock
     private ApplicationEventPublisher applicationEventPublisher;
 
+    @Spy
+    @InjectMocks
+    private ReplicateDemandService replicateDemandService;
+
     @Before
     public void init() {
         MockitoAnnotations.openMocks(this);
     }
 
+    @Test
+    public void shouldRunAskForReplicateAsynchronouslyWhenTriggeredOneTime() throws InterruptedException {
+        ThreadNameWrapper threadNameWrapper = new ThreadNameWrapper();
+        String mainThreadName = Thread.currentThread().getName();
+        doAnswer(invocation -> saveThreadNameThenCallRealMethod(threadNameWrapper, invocation))
+                .when(replicateDemandService).askForReplicate();
+        
+        replicateDemandService.triggerAskForReplicate();
+        TimeUnit.MILLISECONDS.sleep(10);
+        // Make sure askForReplicate method is called 1 time
+        verify(replicateDemandService).askForReplicate();
+        // Make sure askForReplicate method is executed 1 time
```

Comment:
`getLatestBlockNumber` instead of `askForReplicate` ;)
There a few copies of this line throughout the file, I'll let you find them.